testsupport/space.go

```diff
@@ -84,16 +85,23 @@ func WithTierNameAndHashLabel(tierName, hash string) SpaceOption {
 	}
 }
 
+func WithCreationTimestamp(creationTimestamp time.Time) SpaceOption {
+	return func(s *toolchainv1alpha1.Space) {
+		s.CreationTimestamp = metav1.NewTime(creationTimestamp)
+	}
+}
+
 // CreateSpace initializes a new Space object using the NewSpace function, and then creates it in the cluster
 // It also automatically provisions MasterUserRecord and creates SpaceBinding for it
 func CreateSpace(t *testing.T, awaitilities wait.Awaitilities, opts ...SpaceOption) (*toolchainv1alpha1.Space, *toolchainv1alpha1.UserSignup, *toolchainv1alpha1.SpaceBinding) {
 	space := NewSpace(t, awaitilities, opts...)
 	err := awaitilities.Host().CreateWithCleanup(t, space)
 	require.NoError(t, err)
-	space, err = awaitilities.Host().WaitForSpace(t, space.Name, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet())
-	require.NoError(t, err)
 	// we also need to create a MUR & SpaceBinding, otherwise, the Space could be automatically deleted by the SpaceCleanup controller
 	signup, mur, spaceBinding := CreateMurWithAdminSpaceBindingForSpace(t, awaitilities, space, true)
```

Comment:
I wander if we can create a space binding first and then a space...

Reply:
Actually I tried doing that but if I'm not wrong, it doesn't seem possible because `NewSpaceBinding` needs the name of the space https://github.com/codeready-toolchain/toolchain-e2e/blob/210fe8e9eeb171b1bb476197922e1094010ec144/testsupport/spacebinding.go#L43

and that only get's generated by the api server once the space creation was triggered (since some times we use `GenerateName` function)